templates/data/generic_component_data.yaml

```diff
@@ -436,6 +437,9 @@ templates:
                     - jobConfig_presubmit
               - jobConfig:
                   name: post-telemetry-manager-build
+                  branches:
+                    - "^main$"
+                  always_run: true
                   labels:
                     preset-signify-prod-secret: "true"
                   run_if_changed: "^hack/ci/*/.*$|^pkg/*/.*$|^controllers/*/.*$|^charts/*/.*$|^api/*/.*$|^config.yaml$|^Dockerfile$|^(go.mod|go.sum)$|^*/(.*.go|Makefile|.*.sh)|^PROJECT$|^config/*/.*$"
```

Comment:
Remove `run_if_changed`, it's mutual exclusive with `always_run` directive. Your change in data file is not reflected in prowjob definition because of that.

Reply:
run_if_changed removed